### PR TITLE
Add softmin

### PR DESCRIPTION
**Pull request by NoahSchiro**
Adds [softmin](https://docs.pytorch.org/docs/2.13/generated/torch.nn.Softmin.html) alongside softmax.

Often useful in clustering type problems, optimal transport, and other minimization type tasks. 

We also get it nearly for free in terms of LOC

### CODE REVIEW

File: test/backend/test_ops.py [hunk 1]
```diff
@@ -1728,6 +1728,20 @@ def test_log_softmax_other_axis(self):
     helper_test_op([(10,10,10)], lambda x: x.log_softmax(0), atol=1e-7, grad_atol=1e-7)
     helper_test_op([(10,10,10)], lambda x: x.log_softmax(1), atol=1e-7, grad_atol=1e-7)
     helper_test_op([(10,10,10)], lambda x: x.log_softmax(2), atol=1e-7, grad_atol=1e-7)
+  def test_softmin(self):
+    helper_test_op([(45,65)], torch.nn.Softmin(dim=1), Tensor.softmin, atol=1e-7, grad_atol=1e-7)
```
Comment (chenyuxyz) [inline test/backend/test_ops.py:1732]: i think this is fine. for testing compared with `torch.nn.functional.softmin`, and just one is enough. these tests correlate with softmax test anyway

**Comment by geohot** [comment]
are you actually using this, or did you ask an LLM what to add?

**Comment by NoahSchiro** [comment]
Used it a long time ago when investigating using ANNs in gps routing. Not currently using it.

I add it now because it's a feature I frequently see missing from ML frameworks. Here's me adding it to Burn: https://github.com/tracel-ai/burn/pull/2358

It's definitely an infrequently used activation function. Is the utility worth the line count? For you to decide. Every issue / PR from me is merely a suggestion.

No LLM use; arguably worse, I just copy pasted your softmax code / tests.

**Comment by geohot** [comment]
kk, implement chenyu's suggestion and we'll merge
